compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -235,25 +235,46 @@ private static void generateLockForVariable(ClassWriter cw) {
         fv.visitEnd();
     }
 
-    private static void generateStaticInitializer(ClassWriter cw, String className,
-                                                  PackageID packageID, boolean isInitClass,
-                                                  boolean serviceEPAvailable, AsyncDataCollector asyncDataCollector) {
+    private static void generateStaticInitializer(ClassWriter cw, String className, BIRPackage birPackage,
+                                                  boolean isInitClass, boolean serviceEPAvailable,
+                                                  AsyncDataCollector asyncDataCollector,
+                                                  JvmBStringConstantsGen stringConstantsGen) {
         if (!isInitClass && asyncDataCollector.getStrandMetadata().isEmpty()) {
             return;
         }
         MethodVisitor mv = cw.visitMethod(ACC_STATIC, "<clinit>", "()V", null, null);
         if (isInitClass) {
+            setConstantFields(mv, birPackage, className, stringConstantsGen);
```

Comment:
This might get method too large issue if we have around 6000 constants in the module since we initialize all constants in <clinit> method. Please refer JvmBStringConstantsGen class. Shall we test that too?

Reply:
A better way would be to use separate constant gen-class and initialize the constants there without using init class static initializer.

Reply:
We are alrady initilizing at some point, right? So the issue is with ide is not picking it up.

Reply:
Earlier we used to replace the constants with exact values. So constants are not initialized.

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/30648